server/resources/resource_management.go

```diff
@@ -1170,25 +1166,89 @@ func extendFilterByTenantAndDomain(
 	}
 }
 
-func validateAttachmentRelation(
+func validateAttachmentRelationUnderPath(
 	context middleware.Context,
 	policy *schema.Policy,
 	auth schema.Authorization,
-	dataMap map[string]interface{},
-	property *schema.Property,
+	data interface{},
+	resourceSchema *schema.Schema,
+	path string,
 ) error {
-	manager := schema.GetManager()
-	if property.Relation == "" {
+	var currProp *schema.Property
+	currChildren := resourceSchema.Properties
+	dataSet := []interface{}{data}
+
+outer:
+	for _, key := range strings.Split(path, ".") {
+		if key == schema.ItemPropertyID {
+			if currProp == nil {
+				// Invalid path - tried to treat top level object as an array
+				return fmt.Errorf("Invalid path: %s", path)
+			}
+			currProp = currProp.Items
+			currChildren = currProp.Properties
+
+			var newDataSet []interface{}
+			for _, dataPiece := range dataSet {
+				dataPieceArray, _ := dataPiece.([]interface{})
+				newDataSet = append(newDataSet, dataPieceArray...)
+			}
+
+			dataSet = newDataSet
+
+			continue outer
+		}
+
+		for i := range currChildren {
+			if currChildren[i].ID == key {
+				currProp = &currChildren[i]
+				currChildren = currProp.Properties
+
+				for j, dataPiece := range dataSet {
+					dataPieceMap, _ := dataPiece.(map[string]interface{})
+					dataSet[j], _ = dataPieceMap[key]
+				}
+
+				continue outer
```

Comment:
IMHO changing first `continue outer` to `continue`, this one to `childFound = true` and `break` to `if !childFound { break }` would be clearer

Reply:
Refactored into something akin to iterator pattern